bitcoin/src/crypto/ecdsa.rs

```diff
@@ -6,10 +6,14 @@
 
 use core::str::FromStr;
 use core::{fmt, iter};
+use std::convert::TryFrom;
+use alloc::boxed::Box;
 
 use hex::FromHex;
 use internals::write_err;
 use io::Write;
+use std::rc::Rc;
+use std::sync::Arc;
```

Comment:
I don't see a good reason to bring `Rc`/`Arc` here. Why not leave the implementation to the user to deal with the `Rc`/`Arc`?

What am I missing? Glad to be convinced otherwise here.

Reply:
It's pretty natural being able to convert any kind of string and we could optimize the error to carry `Rc`/Arc` directly to avoid cloning the entire string if we want to.